File: testapp/src/main/java/com/appliedrec/mrtd_reader_app/DocumentDetailsActivity.java
package com.appliedrec.mrtd_reader_app;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.appliedrec.mrtd_reader_sample.databinding.ActivityDocumentDetailsBinding;
import com.appliedrec.mrtdreader.MRTDScanResult;

public class DocumentDetailsActivity extends AppCompatActivity {

    MRTDScanResult scanResult;

    public static final String EXTRA_MRTD_SCAN_RESULT = "com.appliedrec.EXTRA_MRTD_SCAN_RESULT";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getIntent() != null) {
            scanResult = getIntent().getParcelableExtra(EXTRA_MRTD_SCAN_RESULT);
        }
        if (scanResult == null) {
            finish();
            return;
        }
        ActivityDocumentDetailsBinding viewBinding = ActivityDocumentDetailsBinding.inflate(getLayoutInflater());
        setContentView(viewBinding.getRoot());
        viewBinding.documentNumber.setText(scanResult.getDocumentNumber());
        viewBinding.personalNumber.setText(scanResult.getPersonalNumber());
        viewBinding.documentCode.setText(scanResult.getDocumentCode());
        viewBinding.issuingState.setText(scanResult.getIssuingState());
        viewBinding.dateOfExpiry.setText(scanResult.getDateOfExpiry());
        viewBinding.primaryIdentifier.setText(scanResult.getPrimaryIdentifier());
        viewBinding.secondaryIdentifiers.setText(TextUtils.join(", ", scanResult.getSecondaryIdentifiers()));
        viewBinding.nationality.setText(scanResult.getNationality());
        viewBinding.gender.setText(scanResult.getGender());
        viewBinding.dateOfBirth.setText(scanResult.getDateOfBirth());
        if (scanResult.getFaceImage() != null) {
            viewBinding.imageView.setImageBitmap(scanResult.getFaceImage());
        } else {
            viewBinding.imageView.setVisibility(View.GONE);
        }
    }
}

File: testapp/src/main/java/com/appliedrec/mrtd_reader_app/DocumentDetailsActivity.kt
package com.appliedrec.mrtd_reader_app

import android.net.Uri
import android.os.Bundle
import android.text.TextUtils
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.appliedrec.mrtd_reader_app.databinding.ActivityDocumentDetailsBinding
import com.appliedrec.mrtdreader.MRTDScanResult
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

class DocumentDetailsActivity : AppCompatActivity() {

    var scanResult: MRTDScanResult.Success? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val viewBinding = ActivityDocumentDetailsBinding.inflate(
            layoutInflater
        )
        setContentView(viewBinding.root)
        val uri: Uri = intent?.data ?: run {
            finish()
            return
        }
        lifecycleScope.launch(Dispatchers.IO) {
            scanResult = ResultFileHelper.readScanResult(uri)
            withContext(Dispatchers.Main) {
                viewBinding.documentNumber.text = scanResult!!.documentNumber
                viewBinding.personalNumber.text = scanResult!!.personalNumber
                viewBinding.documentCode.text = scanResult!!.documentCode
                viewBinding.issuingState.text = scanResult!!.issuingState
                viewBinding.dateOfExpiry.text = scanResult!!.dateOfExpiry
                viewBinding.primaryIdentifier.text = scanResult!!.primaryIdentifier
                viewBinding.secondaryIdentifiers.text =
                    TextUtils.join(", ", scanResult!!.secondaryIdentifiers)
                viewBinding.nationality.text = scanResult!!.nationality
                viewBinding.gender.text = scanResult!!.gender
                viewBinding.dateOfBirth.text = scanResult!!.dateOfBirth
                if (scanResult!!.faceImage != null) {
                    viewBinding.imageView.setImageBitmap(scanResult!!.faceImage)
                } else {
                    viewBinding.imageView.visibility = View.GONE
                }
                viewBinding.signatureVerified.text = if (scanResult!!.signatureVerified) {
                    getString(R.string.yes)
                } else {
                    getString(R.string.no)
                }
                viewBinding.certificateVerified.text = if (scanResult!!.issuerVerified) {
                    getString(R.string.yes)
                } else {
                    getString(R.string.no)
                }
            }
        }
    }
}